MBBSEmu/Logging/Targets/IQueueTarget.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+
+namespace MBBSEmu.Logging.Targets
+{
+    public interface IQueueTarget
+    {
+        object[] Dequeue();
+        IList<object[]> DequeueAll();
+        void Clear();
+    }
```

Comment:
make Generic?

Reply:
C# doesn't support unknown generics like Java (we discussed in Discord), and technically this accomplished the same thing since `object` will need to be boxed to the destination type. My goal was to keep the Targets generic enough as to where a given target can be assigned to any Logger type.